.github/workflows/macos.yml

```diff
@@ -24,51 +24,69 @@ jobs:
       # - for a refresh the key has to change, e.g., hash of a tracked file in the key
       with:
         path: |
-          /usr/local/Caskroom
-          /usr/local/Cellar
-          /usr/local/Frameworks
-          /usr/local/Homebrew
-          /usr/local/bin
-          /usr/local/etc
-          /usr/local/include
-          /usr/local/lib
-          /usr/local/opt
-          /usr/local/sbin
-          /usr/local/var
-          /Users/runner/Library/Caches/Homebrew
+          /usr/local/Cellar/ccache
+          /usr/local/Cellar/fftw
+          /usr/local/Cellar/libomp
+          /usr/local/Cellar/ninja
+          /usr/local/Cellar/numpy
+          /usr/local/Cellar/open-mpi
+          /usr/local/Cellar/openpmd-api
         key: brew-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}
         restore-keys: |
           brew-macos-appleclang-
-    - name: install dependencies
+    - name: install ccache
       run: |
         brew --cache
         set +e
         brew unlink gcc
         brew update
-        brew install --overwrite python
+        brew install ccache --only-dependencies --force
         brew install ccache
+        brew link ccache
+    - name: CCache Cache
+      uses: actions/cache@v3
+      # - once stored under a key, they become immutable (even if local cache path content changes)
+      # - for a refresh the key has to change, e.g., hash of a tracked file in the key
+      with:
+        path: /Users/runner/Library/Caches/ccache
+        key: ccache-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}-${{ hashFiles('cmake/dependencies/AMReX.cmake') }}-${{ github.event.repository.pushed_at }}
+        restore-keys: |
+          ccache-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}-
+          ccache-macos-appleclang-
+    - name: install dependencies
+      run: |
+        brew --cache
+        set +e
+        brew install --overwrite python
+
+        brew install fftw --only-dependencies --force
         brew install fftw
+        brew link fftw
```

Comment:
Why do you need to do this for multiple libraries?

Reply:
Because the packages we use directly are cached with `/usr/local/Cellar/fftw`, the line `brew install fftw` might do nothing. However, it still needs to install the dependencies (which are not cached in case they change over time) so there is `brew install fftw --only-dependencies --force`. Finally, if the package was cached it still needs to be linked so I added `brew link fftw`. This does nothing if the package is not cached and was already installed in `brew install fftw`.